lib/databases/firestore.js

```diff
@@ -0,0 +1,286 @@
+var util = require('util'),
+    _ = require('lodash'),
+    async = require('async'),
+    ConcurrencyError = require('../concurrencyError'),
+    gcFirestore = require('@google-cloud/firestore'),
```

Comment:
Can you use `Repository.use('@google-cloud/firestore')` instead of `require('@google-cloud/firestore')` ?